src/controllers/EmailsControllers.ts

```diff
@@ -209,34 +273,77 @@ export function sendMail(
 > {
   const controller = sendMailController(config, mailTrasporter, browserEngine);
   // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-  return async req => {
-    const errorOrBody = NotificationEmailRequest.decode(req.body);
-    if (E.isLeft(errorOrBody)) {
-      const error = errorOrBody.left;
-      return ResponseErrorFromValidationErrors(NotificationEmailRequest)(error);
-    }
-    const body = errorOrBody.right;
+  return async req =>
+    pipe(
+      req.body,
+      NotificationEmailRequest.decode,
+      E.mapLeft(async e =>
+        ResponseErrorFromValidationErrors(NotificationEmailRequest)(e)
+      ),
+      E.bindTo("body"),
+      E.bind("clientId", () =>
+        pipe(
+          getClientId(req),
+          E.mapLeft(e => headerValidationErrorHandler(e))
+        )
+      ),
+      E.chainFirst(({ body, clientId }) =>
+        pipe(
+          validTemplateIdGivenClientConfig(
+            config[clientId as NotificationsServiceClientEnum],
+            body.templateId
+          ),
+          E.mapLeft(async e => ResponseErrorValidation(e.name, e.message))
+        )
+      ),
+      E.fold(
+        e => e,
+        ({ body, clientId }) =>
+          controller({
+            body,
+            "X-Client-Id": clientId
+          })
+      )
+    );
+}
 
-    const maybeClientId = getClientId(req);
-    if (E.isLeft(maybeClientId)) {
-      const error = maybeClientId.left;
-      return headerValidationErrorHandler(error);
-    }
-    const clientId = maybeClientId.right as NotificationsServiceClientEnum;
+export const addRetryQueueListener = (
```

Comment:
why define it in the controller?

Reply:
maybe in a dedicated file, what do you think?

Reply:
I'm ok with that, the question was more about the naming 😄 (and the parent directory)

Reply:
`src/queues/RetryListener.ts` or something like that?